core/src/env/traits.rs

```diff
@@ -18,12 +18,13 @@ use crate::{
     memory::vec::Vec,
     storage::Key,
 };
+use core::hash::Hash;
 use parity_codec::Codec;
 
 /// The environmental types usable by contracts defined with pDSL.
 pub trait EnvTypes {
     /// The type of an address.
-    type AccountId: Codec + PartialEq + Eq;
+    type AccountId: Codec + PartialEq + Eq + Hash + Copy;
```

Comment:
Why does it need to be `Hash` and `Copy`? We need to take care to not be more restrictive than Substrate.

Reply:
I agree, these are requirements only for using `AccountId` with a `HashMap` (e.g. ERC20 sample), which is not a strict requirement. Will remove